components/shared/DropdownButton/DropdownButton.tsx

```diff
@@ -0,0 +1,91 @@
+import { Dropdown, Image } from "../../bootstrap"
+import classNames from "classnames"
+import styled from "styled-components"
+import styles from "./dropdownButton.module.css"
+import { CarotIcon } from "./CarotIcon"
+import { Fragment } from "react"
+
+export type DropdownButtonProps = {
+  title: string
+  children: Array<string>
+  variant: string
+  styling?: stylingProps
+}
+
+export type stylingProps = {
+  width?: { desktop: string; mobile: string }
+  height?: string
+  fontSize?: string
+}
+
+const StyledDropdown = styled(Dropdown)`
+  .toggle {
+    width: ${props => props.changewidth?.desktop ?? "418px"};
+    height: ${props => props.changeheight ?? "40px"};
+    font-size: ${props => props.changefontsize ?? "1rem"};
+  }
+
+  .menu {
+    width: ${props => props.changewidth?.desktop ?? "418px"};
+
+    font-size: ${props =>props.changefontsize ?? "1rem"};
+  }
+
+  .carotIcon {
+    fill: #ffffff;
+  }
+
+  @media only screen and (max-width: 768px) {
+    .toggle {
+      width: ${props => props.changewidth?.mobile ?? "418px"};
+    }
+    .menu {
+      width: ${props => props.changewidth?.mobile ?? "418px"};
+    }
+  }
+`
+
+export function DropdownButton({
+  title,
+  children,
+  variant,
+  styling
+}: DropdownButtonProps): JSX.Element {
+  return (
+    <StyledDropdown
+      changewidth={styling?.width}
+      changeheight={styling?.height}
+      changefontsize={styling?.fontSize}
+      variant={variant}
+    >
+      <Dropdown.Toggle
+        className={classNames(styles.dropdownToggle, "shadow-none", "toggle")}
+        variant={variant}
```

Comment:
```suggestion
import { CarotIcon } from "./CarotIcon"

export type DropdownButtonProps = {
  title: string
  children: Array<string>
  variant: string
  className?: string
}

export type stylingProps = {
  width?: { desktop: string; mobile: string }
  height?: string
  fontSize?: string
}

const StyledDropdown = styled(Dropdown)`
  .dropdown-toggle {
    height: 40px;
  }

  .dropdown,
  .dropdown-toggle,
  .dropdown-menu {
    width: 100%;
  }

  .carotIcon {
    fill: #ffffff;
  }
`

export function DropdownButton({
  title,
  children,
  variant,
  className
}: DropdownButtonProps): JSX.Element {
  return (
    <StyledDropdown variant={variant} className={className}>
      <Dropdown.Toggle
        className={classNames(styles.dropdownToggle, "shadow-none", "toggle")}
        variant={variant}
```

I really like that you're thinking about how we can ensure consistent styling across the site for each component! I think it's good to allow customization where necessary, and to make use of the existing features of bootstrap when we can. So here, rather than define your own class name for toggle and menu, and styling overrides for width/height, you can reuse the bootstrap classnames. This makes it easier for people to customize the dropdown button because they can refer to the existing bootstrap docs. and wrap this component in another `styled`, to override classes as you do here.

In general, the react version of a bootstrap component uses the css from the corresponding component in the base bootstrap library (ex [dropdown](https://getbootstrap.com/docs/4.1/components/dropdowns/)). You can also inspect the page elements to get the classname. For this, the toggle uses `dropdwon-toggle` and the menu use `dropdown-menu`.

As for the styling overrides, it seems the distinction between bootstrap's dropdown and ours is that ours has the toggle and the menu as the same width, as well as different default values for height and the caret. By setting the width on the toggle, menu, and item to 100%, the whole component will be the width of its container. Then, we can use the bootstrap grid layout to set the size of the dropdown (see Faq.tsx). We can hard-code the caret and height in this styled component, as users can override this using another `styled` if they need to.

Reply:
A few things on bootstrap class names being used here: 

- styling overrides can only happen IF the person copy and pastes the entire dropdown into their file and then styles it anew. This would conflict with "don't repeat yourself" coding principles (DRY) and would defeat the point of having a shared component. 
- I can change the bootstrap class names, this is actually what I had originally but changed them after a prev reviewer said there was 'style conflicts' - I'm happy to change it back
- The carot was made dynamic because it needs to change in color if color is customized 
- to address the points in your last comment - that method is something I tried but did not work- I have clarified this in more detail in your last comment on my code (if you scroll down you'll see it)

Reply:
A few things on bootstrap class names being used here: 

- styling overrides can only happen IF the person copy and pastes the entire dropdown into their file and then styles it anew. This would conflict with "don't repeat yourself" coding principles (DRY) and would defeat the point of having a shared component. When the dropdown is imported, the return bootstrap dropdown (toggle, menu, items) are unreachable. It is not possible to change them via modifying bootstrap classNames.
- I can change the bootstrap class names, this is actually what I had originally but changed them after a prev reviewer said there was 'style conflicts' - I'm happy to change it back
- The carot was originally made dynamic because it needs to change in color if color is customized - I went back and forth on this- I'm leaving it where it is right now so that if change of color is needed in the future (UX library is ongoing)- it'll be easy to make it dynamic
- to address the points in your last comment - that method is something I tried but did not work- I have clarified this in more detail in your last comment on my code (if you scroll down you'll see it)
- if you'd like, feel free to code an example of customizing the width of the dropdown based on the method you suggested...

Reply:
- styled-components works by generating classes and passing them into the wrapped component as `className`, so the trick is to add `className` to DropdownButton and pass that into the Dropdown component. [This commit](https://github.com/codeforboston/maple/commit/8c44e79c25b304ef6c579a6ef48e8c0a68356ec6) demonstrates overriding styles of `DropdownButton` by targeting the bootstrap class names ([deployed page](https://maple-dev-git-fork-alexjball-pr-1354-sug-2c0ec7-maple-testimony.vercel.app/about/faq-page)).
- I'm not sure what you mean by change the bootstrap class names. this should not modify the class names bootstrap uses (it should not need to change bootstrap.scss). I just mean that bootstrap already adds standard class names to each of the components that make up a dropdown, so you can generally use those for your css selectors, rather than add additional ones. In the ss below, you can see the bootstrap-defined `dropdown-item` class alongside the `item` class added here.
- The suggestions I left originally size the dropdown based on its container.

Reply:
Okay so it's an issue of "tomayto, tomahto." My code functions as expected and it is natural that coders will come up with different ways of coding things...